WeScan/Extensions/CIRectangleFeature+Utils.swift

```diff
@@ -23,28 +23,10 @@ extension CIRectangleFeature {
     ///   - rectangleFeature: The other rectangle to compare this instance with.
     /// - Returns: True if the given rectangle is within the given distance of this rectangle instance.
     func isWithin(_ distance: CGFloat, ofRectangleFeature rectangleFeature: CIRectangleFeature) -> Bool {
-        
-        let topLeftRect = topLeft.surroundingSquare(withSize: distance)
-        if !topLeftRect.contains(rectangleFeature.topLeft) {
-            return false
-        }
-        
-        let topRightRect = topRight.surroundingSquare(withSize: distance)
-        if !topRightRect.contains(rectangleFeature.topRight) {
-            return false
-        }
-        
-        let bottomRightRect = bottomRight.surroundingSquare(withSize: distance)
-        if !bottomRightRect.contains(rectangleFeature.bottomRight) {
-            return false
-        }
-
-        let bottomLeftRect = bottomLeft.surroundingSquare(withSize: distance)
-        if !bottomLeftRect.contains(rectangleFeature.bottomLeft) {
-            return false
-        }
-        
-        return true
+        return [\CIRectangleFeature.topLeft, \CIRectangleFeature.topRight,
+                \CIRectangleFeature.bottomLeft, \CIRectangleFeature.bottomRight]
+               .map { self[keyPath: $0].surroundingSquare(withSize: distance).contains(rectangleFeature[keyPath: $0]) }
+               .reduce(true, { $0 && $1})
```

Comment:
For readability, it might be better to write out the parameters here.

Reply:
If you mean map, it has the only parameter and it is sequence's element. Don't see reason giving it name. Name not only will not improve readability it will make code of this map multiline which in turn worsen readability.

Same applies to reduce where anonymous parameters play really good.

If you insist giving parameters names, could give name `keypath` for parameter of map closure.

Can't thing of anything better then `flag0` and `flag1` for parameters of reduce closure. Don't see how is this better then `$0` and `$1`.

Reply:
Thanks for this great explanation, I can understand your decisions here. All good!